src/intersections/boxes.jl

```diff
@@ -51,16 +51,16 @@ function intersecttype(f::Function, r::Ray{Dim,T}, b::Box{Dim,T}) where {Dim,T}
   lo, up = coordinates.(extrema(b))
   orig = coordinates(origin(r))
 
-  tmin = zero(T)
-  tmax = typemax(T)
+  tmin = zero(T) / oneunit(T)
+  tmax = typemax(T) / oneunit(T)
 
   # check for intersection with slabs along with each axis
   for i in 1:Dim
     imin = (lo[i] - orig[i]) * invdir[i]
     imax = (up[i] - orig[i]) * invdir[i]
 
     # swap variables if necessary
-    invdir[i] < zero(T) && ((imin, imax) = (imax, imin))
+    invdir[i] < zero(T)^3 && ((imin, imax) = (imax, imin))
```

Comment:
@juliohm atol(T)^2 does not seem to work. For example, if atol(T) is 1e-10 mm, then atol(T)^2 will make 1e-20 mm^2. However, the purpose of using atol is to avoid floating-point errors. Although the unit of atol(T)^2 is correct, it also changes the value, which is not what we want.